src/index.ts

```diff
@@ -64,32 +111,20 @@ export async function install(
 
   const depObject = {};
   for (const dep of arrayOfDeps) {
-    const depLoc = path.join(cwd, 'node_modules', dep);
-    if (!fs.existsSync(depLoc)) {
-      logError(
-        `dependency "${dep}" not found in your node_modules/ directory. Did you run npm install?`,
-      );
-      return;
-    }
-    const depManifestLoc = path.join(cwd, 'node_modules', dep, 'package.json');
-    const depManifest = require(depManifestLoc);
-    if (!depManifest.module) {
-      if (isWhitelist) {
-        logError(`dependency "${dep}" has no ES "module" entrypoint.`);
-        console.log(
-          '\n' +
-            chalk.italic(
-              `Tip: Find modern, web-ready packages at ${chalk.underline(
-                'https://pikapkg.com/packages',
-              )}`,
-            ) +
-            '\n',
-        );
+    try {
+      const depName = getWebDependencyName(dep);
+      const depLoc = resolveWebDependency(dep);
+      depObject[depName] = depLoc;
+    } catch (err) {
+      // An error occurred! Log it.
+      logError(err.message || err);
+      if (err.hint) {
+        console.log(err.hint);
+      }
+      if (!skipFailures) {
```

Comment:
whenever there's no explicit webDependencies: we forgive all types of Error. is this acceptable? like, if Node was trying to tell you you've run out of disk space we'd catch that and keep hammering it.

Reply:
Haha I know, I actually have a lot of trouble with throwing "expected" errors as well. We could turn `ErrorWithHint` into some `ExpectedError` class (with an optional hint) so that we could only report and skip expected pika-related errors. Or we could create an error return type and remove the try/catch entirely.

Reply:
The other issue here is: do we even want an implicit no-webDependencies flow? I think it's useful to give new users a "zero-config mode", but could we be better at encouraging it as the main flow?

Reply:
> TS doesn't even allow you to type exceptions

maybe it's better to use duck-typing.

```js
try {
//…
throw Object.assign(new Error('the message'), {
  hint: 'the hint'
});
//…
} catch(err) {
  if (err.hint) console.warn(hint)
}
```

Reply:
> `ExpectedError` class (with an optional hint)

certainly there's no obvious hierarchy, so making something as specific as an ErrorWithHint subclass is wrong; hints are only one of many things we may want to decorate onto an error.

ExpectedError with optional hint (and optional other things) is less brittle; I guess it works out to be a union of many different error subclasses.

Reply:
one way to support a variety of decorators without using subclasses is:

```ts
interface Person {
    firstName: string;
    lastName: string;
}

interface ErrorWithHint extends Error {
  hint: string;
}

function hasHint(err: Error) err is ErrorWithHint {
    return err.hint; // you could check typeof rather than just truthiness
}

//…
} catch(err) {
  if (hasHint(err)) {
    // inside this block, the error is cast to ErrorWithHint
    console.warn(hint)
  }
}
```

Reply:
> The other issue here is: do we even want an implicit no-webDependencies flow?

I think it's questionable. There are maaany packages in my node_modules folder; automatically attempting to generate an ES module for each of them would not be useful to me.

If the user is happy to fill in `dependencies` manually, then they're probably happy to fill in `webDependencies`, too.

Moreover: the auto-config didn't detect subpackages like `preact/hooks` for me, so I was very confused about how to get from no config at all to a very specific config.

Reply:
So one thing that's important to keep in mind with all of this is that it only creates `web_modules/` files for your top-level dependencies (those mentioned in your package.json `dependencies` object explicitly)  and not every package in your `node_modules/` dir.